diff --git a/flake.lock b/flake.lock
index a63d3c7..1e42ef4 100644
--- a/flake.lock
+++ b/flake.lock
@@ -506,6 +506,7 @@
         "nixos-generators": "nixos-generators",
         "nixos-hardware": "nixos-hardware",
         "nixpkgs": "nixpkgs",
+        "srta-ldpi": "srta-ldpi",
         "systems": "systems",
         "treefmt-nix": "treefmt-nix"
       }
@@ -547,6 +548,30 @@
         "url": "https://spectrum-os.org/git/spectrum"
       }
     },
+    "srta-ldpi": {
+      "inputs": {
+        "flake-parts": [
+          "flake-parts"
+        ],
+        "nixpkgs": [
+          "nixpkgs"
+        ]
+      },
+      "locked": {
+        "lastModified": 1732795704,
+        "narHash": "sha256-3SKnnX874pfrs1soJ99DRf+Zs4oItmuKw7nzT41Qf+w=",
+        "owner": "tiiuae",
+        "repo": "srta-ldpi",
+        "rev": "1b094ef4629d438bab38d0388777f8947392bb65",
+        "type": "github"
+      },
+      "original": {
+        "owner": "tiiuae",
+        "repo": "srta-ldpi",
+        "rev": "1b094ef4629d438bab38d0388777f8947392bb65",
+        "type": "github"
+      }
+    },
     "systems": {
       "locked": {
         "lastModified": 1681028828,
diff --git a/flake.nix b/flake.nix
index ed84d10..0a21fdc 100644
--- a/flake.nix
+++ b/flake.nix
@@ -157,6 +157,14 @@
       };
     };
 
+    srta-ldpi = {
+      url = "github:tiiuae/srta-ldpi/1b094ef4629d438bab38d0388777f8947392bb65";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-parts.follows = "flake-parts";
+      };
+    };
+
     ci-test-automation = {
       url = "github:tiiuae/ci-test-automation";
       inputs = {
diff --git a/modules/microvm/virtualization/microvm/netvm.nix b/modules/microvm/virtualization/microvm/netvm.nix
index 8786121..ca61149 100644
--- a/modules/microvm/virtualization/microvm/netvm.nix
+++ b/modules/microvm/virtualization/microvm/netvm.nix
@@ -12,6 +12,10 @@ let
     imports = [
       inputs.impermanence.nixosModules.impermanence
       inputs.self.nixosModules.givc-netvm
+      
+      # Importing the srta-ldpi module
+      inputs.srta-ldpi.nixosModules.ldpi
+      
       (import ./common/vm-networking.nix {
         inherit
           config
@@ -114,6 +118,7 @@ let
           microvm = {
             # Optimize is disabled because when it is enabled, qemu is built without libusb
             optimize.enable = false;
+            mem = 1024;
             hypervisor = "qemu";
             shares = [
               {
